floss/language/go/extract.py

```diff
@@ -0,0 +1,318 @@
+    .text:00000000004736F0 4C 8D 05 84 47 03 00            lea     r8, unk_4A7E7B
+
+    386:
+    .text:004806D2 8D 05 EC 1D 4A 00                       lea     eax, unk_4A1DEC
+
+    """
+    text_segment_xrefs = list()
+
+    if arch == "amd64":
+        text_regex = re.compile(b"(\x48|\x4C)\x8D(?=.(?P<offset>....))", re.DOTALL)
+        for match in text_regex.finditer(text_segment_data):
+            offset = struct.unpack("<I", match.group("offset"))[0]
+            address = text_segment_va + match.start() + offset + 7 + pe.OPTIONAL_HEADER.ImageBase
+            if start_address <= address <= end_address:
+                text_segment_xrefs.append(address)
+    else:
+        text_regex = re.compile(b"\x8D(?=.(?P<offset>....))", re.DOTALL)
+        for match in text_regex.finditer(text_segment_data):
+            offset = struct.unpack("<I", match.group("offset"))[0]
+            address = offset
+            if start_address <= address <= end_address:
+                text_segment_xrefs.append(address)
+
+    return text_segment_xrefs
+
+
+def xrefs_in_rdata_data_segment(section_data, start_address, end_address, arch) -> List[int]:
+    """
+    Find cross-references to a string in the .rdata segment.
+    All cross-references are of the form:
+    00000000004C9D00  19 8C 4A 00 00 00 00 00  0A 00 00 00 00 00 00 00  ..J.............
+    """
+
+    if arch == "amd64":
+        size = 0x10
+        fmt = "<QQ"
+    else:
+        size = 0x8
+        fmt = "<II"
+
+    xrefs_in_rdata_data_segment = list()
+
+    for addr in range(0, len(section_data) - size // 2, size // 2):
+        curr = section_data[addr : addr + size]
+        s_off, s_size = struct.unpack_from(fmt, curr)
+
+        if not (1 <= s_size < 128):
+            continue
+
+        if start_address <= s_off <= end_address:
+            xrefs_in_rdata_data_segment.append(s_off)
+
+    return xrefs_in_rdata_data_segment
+
+
+def split_string_by_indices(string, indices, max_xref_string_start, max_xref_string_end):
+    """Split a string into parts by indices."""
+    parts = []
+    previous_index = 0
+
+    for index in indices:
+        index -= max_xref_string_start
+        if index > max_xref_string_end:
+            break
+        if index < 0:
+            continue
+        parts.append((index, string[previous_index:index]))
+        previous_index = index
+
+    parts.append((previous_index, string[previous_index:]))
+
+    return parts
+
+
+def count_elements_between(numbers, start_number, end_number) -> int:
+    start_index = 0
+    end_index = len(numbers) - 1
+
+    while start_index <= end_index:
+        mid_index = (start_index + end_index) // 2
+        if numbers[mid_index] < start_number:
+            start_index = mid_index + 1
+        else:
+            end_index = mid_index - 1
+
+    while end_index < len(numbers) - 1 and numbers[end_index + 1] <= end_number:
+        end_index += 1
+
+    count = end_index - start_index + 1
+    return count
```

Comment:
would you add a docstring comment to this routine, maybe with an example, that describes what this function does? i can't quite figure it out with the current name and logic.